torch/csrc/jit/script/compilation_unit.h

```diff
@@ -252,6 +256,41 @@ struct TORCH_API CompilationUnit {
     functions_.clear();
   }
 
+  /**
+   * Register a class as being owned by this compilation unit.
+   */
+  void register_class(ClassTypePtr classType) {
+    classes_.push_back(std::move(classType));
+  };
+
+  ClassTypePtr get_class(const c10::QualifiedName& name) const {
+    for (const auto& cls : classes_) {
+      if (cls->qualname() == name.qualifiedName()) {
+        return cls;
+      }
+    }
+    return nullptr;
+  }
+
+  /**
```

Comment:
I feel like this should be in a python-only compilation unit, for clarity.

Reply:
Maybe python_sugared_value.cpp

Reply:
Unfortunately because of dependency stuff I can't move this cleanly; I'll follow up with the change in another PR